src/code/z_bgcheck.c

```diff
@@ -3826,129 +3826,128 @@ u32 SurfaceType_GetData(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId,
 }
 
 /**
- * SurfaceType return CamData Index
+ * SurfaceType Get index of bgCamData
  */
-u32 SurfaceType_GetCamDataIndex(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
-
+u32 SurfaceType_GetBgCamDataId(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
     return SurfaceType_GetData(colCtx, poly, bgId, 0) & 0xFF;
 }
 
 /**
- * CamData return cameraSType
+ * BgCamData Get setting of bgCamData
  */
-u16 func_80041A4C(CollisionContext* colCtx, u32 camId, s32 bgId) {
-    u16 result;
+u16 SurfaceType_GetBgCamDataSetting(CollisionContext* colCtx, u32 bgCamDataId, s32 bgId) {
```

Comment:
I think I understand the gist of it, my main question is how would `SurfaceType_GetBgCamDataSetting` be differentiated from `WaterBox_GetBgCamDataSetting`?

Reply:
`WaterBox_GetBgCamDataSetting` uses both waterbox and bgcamdata stuff, I guess it should keep the WaterBox prefix to help "isolate" the waterbox system (also for consistency with eg `WaterBox_GetBgCamDataId` which definitely can keep the WaterBox prefix since it only uses waterbox data)

if you do want to make both `SurfaceType_GetBgCamDataSetting` and `WaterBox_GetBgCamDataSetting` use the same prefix though I'd suggest `BgCamData_GetSetting` and `BgCamData_GetWaterBoxSetting`, but I like different prefixes better

Reply:
I'm not entirely sure if the `SurfaceType_` prefix should be removed, would be open to another opinion!

Reply:
Thinking about this a little more, since the prefix `SurfaceType_` deals a lot more than just camera, I feel renaming or removing this prefix should be saved for a future `bgcheck` document pass instead of putting it inside this `camera` document pass, but we can take this to discord to discuss further if you'd like